ci/gpu/build.sh

```diff
@@ -79,73 +79,148 @@ $CC --version
 
+    
+    logger "Python pytest for cuml..."
+    cd $WORKSPACE/python
 
-################################################################################
-# TEST - Run notebook tests
-################################################################################
+    pytest --cache-clear --basetemp=${WORKSPACE}/cuml-cuda-tmp --junitxml=${WORKSPACE}/junit-cuml.xml -v -s -m "not memleak" --durations=50 --timeout=300 --ignore=cuml/test/dask --ignore=cuml/raft
 
-${WORKSPACE}/ci/gpu/test-notebooks.sh 2>&1 | tee nbtest.log
-python ${WORKSPACE}/ci/utils/nbtestlog2junitxml.py nbtest.log
+    timeout 7200 sh -c "pytest cuml/test/dask --cache-clear --basetemp=${WORKSPACE}/cuml-mg-cuda-tmp --junitxml=${WORKSPACE}/junit-cuml-mg.xml -v -s -m 'not memleak' --durations=50 --timeout=300"
 
-################################################################################
-# TEST - Run GoogleTest for ml-prims
-################################################################################
 
-logger "Run ml-prims test..."
-cd $WORKSPACE/cpp/build
-GTEST_OUTPUT="xml:${WORKSPACE}/test-results/prims/" ./test/prims
+    ################################################################################
+    # TEST - Run notebook tests
+    ################################################################################
 
-################################################################################
-# TEST - Run GoogleTest for ml-prims, but with cuda-memcheck enabled
-################################################################################
+    ${WORKSPACE}/ci/gpu/test-notebooks.sh 2>&1 | tee nbtest.log
+    python ${WORKSPACE}/ci/utils/nbtestlog2junitxml.py nbtest.log
 
-if [ "$BUILD_MODE" = "branch" ] && [ "$BUILD_TYPE" = "gpu" ]; then
-    logger "GoogleTest for ml-prims with cuda-memcheck enabled..."
+    ################################################################################
+    # TEST - Run GoogleTest for ml-prims
+    ################################################################################
+
+    logger "Run ml-prims test..."
     cd $WORKSPACE/cpp/build
-    python ../scripts/cuda-memcheck.py -tool memcheck -exe ./test/prims
-fi
+    GTEST_OUTPUT="xml:${WORKSPACE}/test-results/prims/" ./test/prims
+
+    ################################################################################
+    # TEST - Run GoogleTest for ml-prims, but with cuda-memcheck enabled
+    ################################################################################
+
+    if [ "$BUILD_MODE" = "branch" ] && [ "$BUILD_TYPE" = "gpu" ]; then
+        logger "GoogleTest for ml-prims with cuda-memcheck enabled..."
+        cd $WORKSPACE/cpp/build
+        python ../scripts/cuda-memcheck.py -tool memcheck -exe ./test/prims
+    fi
+else
+    #Project Flash
+    export LIBCUML_BUILD_DIR="$WORKSPACE/ci/artifacts/cuml/cpu/conda_work/cpp/build"
+    export LD_LIBRARY_PATH="$LIBCUML_BUILD_DIR:$LD_LIBRARY_PATH"
```

Comment:
Given this new workflow the gpu jobs will now get the `libcuml.so` and test executable artifacts from the conda-build cpu jobs, right? I ask since that means that then the `ci/local` scripts will not reproduce this new workflow, so I was wondering if there are plans in motion already to update them?

Reply:
I haven't considered the effect these changes has on the local scripts. I will have to give that some thought.